packages/go_router/lib/src/information_provider.dart

```diff
@@ -172,6 +172,33 @@ class GoRouteInformationProvider extends RouteInformationProvider
     );
   }
 
+  /// Relatively go to [relativeLocation].
+  void goRelative(String relativeLocation, {Object? extra}) {
+    assert(
+      !relativeLocation.startsWith('/'),
+      "Relative locations must not start with a '/'.",
+    );
+
+    final Uri currentUri = value.uri;
+    Uri newUri = Uri.parse(
```

Comment:
Why not just modify the path and use `currentUri.replace(path: newPath)`
This way you don't need to worry about other thing such as host/scheme/fragment/queryparameters

Reply:
Gotcha. I move this in path_utils as `concatenateUris`